src/main/java/pl/marcin/project/tomtomgeoservice/service/GeoSerwis.java

```diff
@@ -11,23 +14,28 @@
 import pl.marcin.project.tomtomgeoservice.routingmodel.RouteType;
 import pl.marcin.project.tomtomgeoservice.routingmodel.TravelMode;
 
-
+@Service
 public class GeoSerwis {
-    private final String baseUrl = "https://api.tomtom.com";
-    private WebClient webClient = WebClient.create(baseUrl);
-    GeocodingService geocodingService = new GeocodingService(webClient);
-    RouteService routeService = new RouteService(webClient);
-    ObjectMapperConverter objectMapperConverter = new ObjectMapperConverter();
-
-    public GeocodingAnswer getLocationFromAddress(AddressData addressData) throws Exception {
-        String coordinates = geocodingService.getCoordinates(addressData).block();
+
+    private final MapServiceFacade mapServiceFacade;
+    private final ObjectMapperConverter objectMapperConverter;
+
+    @Autowired
+    public GeoSerwis(MapServiceFacade mapServiceFacade, ObjectMapperConverter objectMapperConverter) {
+        this.mapServiceFacade = mapServiceFacade;
+        this.objectMapperConverter = objectMapperConverter;
+    }
+
+
+    private GeocodingAnswer getLocationFromAddress(AddressData addressData) throws Exception {
```

Comment:
can it throw some more specific `Exception` perhaps?

Reply:
can we handle the exception in this method too? I think a service, in our case, could hide any potential work/errors - _or_ we could throw a more meaningful exception in such a case. `Exception` is too vague ;) it can be a variety of resons  - we could handle any potential `Exceptions` within this method and, if needed, throw more specific ones `IllegalArgument` or `IllegalState`